extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php

```diff
@@ -36,134 +33,170 @@ public static function onBeforePageDisplay() {
 
 	/**
 	 * Checks arguments for errors.
+	 * @param $modelData
 	 * @param array $args
-	 * @param string errorMessage Return parameter with the proper error message to show. Disregard if return is false
+	 * @internal param $errorMessage /Return parameter with the proper error message to show.
+	 * Disregard if return is false.
 	 * @return true if ok, false if error
 	 */
-	private function checkArguments( array $args, $modelData, &$errorMessage ) {
+	private function checkArguments( $modelData, array $args, &$errorMessage ) {
 		// mostrecent must be bool
-		if ( isset( $args['mostrecent'] ) &&
-			$args['mostrecent'] !== 'true' &&
-			$args['mostrecent'] !== 'false'
+		if ( isset( $args[ 'mostrecent' ] ) &&
+			$args[ 'mostrecent' ] !== 'true' &&
+			$args[ 'mostrecent' ] !== 'false'
 		) {
-			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error', 'mostrecent' )->plain() .
-				wfMessage( 'embeddable-discussions-parameter-error-boolean' )->plain();
+			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error',
+				'mostrecent',
+				wfMessage( 'embeddable-discussions-parameter-error-boolean' )->plain()
+			)->plain();
 
 			return false;
 		}
 
 		// size must be integer in range
-		if ( isset( $args['size'] ) ) {
-			$size = $args['size'];
+		if ( isset( $args[ 'size' ] ) ) {
+			$sizeArg = $args[ 'size' ];
+			$size = ctype_digit( $sizeArg ) ? intval( $sizeArg ) : $sizeArg;
 
-			if ( !ctype_digit( $size ) ||
-				intval( $size ) > self::ITEMS_MAX ||
-				intval( $size ) < self::ITEMS_MIN
+			if ( !is_int ( $size ) ||
+				$size > self::ITEMS_MAX ||
+				$size < self::ITEMS_MIN
 			) {
-				// TODO: Refactor error messages to avoid direct concatenation, see JPN-657
-				$errorMessage = wfMessage( 'embeddable-discussions-parameter-error', 'size' )->plain() .
-					wfMessage( 'embeddable-discussions-parameter-error-range',
-						self::ITEMS_MIN , self::ITEMS_MAX )->plain();
+				$errorMessage = wfMessage( 'embeddable-discussions-parameter-error',
+					'size',
+					wfMessage( 'embeddable-discussions-parameter-error-range', self::ITEMS_MIN, self::ITEMS_MAX )->plain()
+				)->plain();
 
 				return false;
 			}
 		}
 
 		// columns must be integer in range
-		if ( isset( $args['columns'] ) ) {
-			$columns = $args['columns'];
+		if ( isset( $args[ 'columns' ] ) ) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "columns" 6 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)